### PR TITLE
services.md 

### PR DESCRIPTION
**Pull request by aevesdocker**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Closes https://github.com/compose-spec/compose-spec/issues/347, https://github.com/compose-spec/compose-spec/issues/387, https://github.com/compose-spec/compose-spec/issues/394. https://github.com/docker/docs/issues/13361, https://github.com/compose-spec/compose-spec/issues/359

### CODE REVIEW

File: 05-services.md [hunk 1]
```diff
@@ -1243,18 +1223,18 @@ _DEPRECATED: use [deploy.reservations.memory](deploy.md#memory)_
...
 
-Default value is platform specific.
+The default value is platform specific.
 
 ### memswap_limit
 
-`memswap_limit` defines the amount of memory container is allowed to swap to disk. This is a modifier
-attribute that only has meaning if `memory` is also set. Using swap allows the container to write excess
+`memswap_limit` defines the amount of memory the container is allowed to swap to disk. This is a modifier
+attribute that only has meaning if [`memory`](#memor) is also set. Using swap allows the container to write excess
```
Comment (aevesdocker) [inline 05-services.md:1237]: TODO: clarify, and is this to deploy memory?

File: 05-services.md [hunk 2]
```diff
@@ -1344,19 +1328,21 @@ ports:
...
 The long form syntax allows the configuration of additional fields that can't be
 expressed in the short form.
 
-- `target`: the container port
-- `published`: the publicly exposed port. Can be set as a range using syntax `start-end`, so it is defined as a string, then actual port should be assigned within this range based on available ports.
-- `host_ip`: the Host IP mapping, unspecified means all network interfaces (`0.0.0.0`)
-- `protocol`: the port protocol (`tcp` or `udp`), unspecified means any protocol
-- `mode`: `host` for publishing a host port on each node, or `ingress` for a port to be load balanced.
+- `target`: The container port
+- `published`: The publicly exposed port. Can be set as a range using syntax `start-end`, so it is defined as a string, then actual port should be assigned within this range based on available ports.
```
Comment (aevesdocker) [inline 05-services.md:1342]: TODO: Clarify meaning

File: 05-services.md [hunk 3]
```diff
@@ -1375,40 +1361,54 @@ ports:
...
+   If there is no cached image, a failure is reported.
 * `missing`: Compose pulls the image only if it's not available in the platform cache.
-   This should be the default option if you are not also using the Compose Build specification.
-  `if_not_present` should be considered an alias for this value for backward compatibility
-* `build`: Compose builds the image. Compose rebuilds the image if already present.
+   This is the default option if you are not also using the [Compose Build Specification](build.md).
+  `if_not_present` is considered an alias for this value for backward compatibility.
+* `build`: Compose builds the image. Compose rebuilds the image if it's already present.
 
 If `pull_policy` and `build` are both present, Compose builds the image by default. Compose may override this behavior in the toolchain.
```
Comment (aevesdocker) [inline 05-services.md:1398]: TODO: Clarify last sentence

File: 05-services.md [hunk 4]
```diff
@@ -602,7 +588,7 @@ When both `env_file` and `environment` are set for a service, values set by `env
 
 ### expose
```
Comment (aevesdocker) [inline 05-services.md:589]: TODO: Check https://github.com/compose-spec/compose-spec/issues/293

File: 05-services.md [hunk 5]
```diff
@@ -493,24 +481,24 @@ dns_search:
 
 ### domainname
```
Comment (aevesdocker) [inline 05-services.md:482]: TODO: Add more for https://github.com/compose-spec/compose-spec/issues/358

File: 05-services.md (hunk 4 above)
Comment (aevesdocker) [reply 05-services.md:589]: Check https://github.com/compose-spec/compose-spec/issues/359

File: 05-services.md [hunk 6]
```diff
@@ -1480,13 +1484,13 @@ the service's containers.
   `/run/secrets/` in the service's task containers. Default value is USER running container.
```
Comment (aevesdocker) [inline 05-services.md:1484]: Check https://github.com/compose-spec/compose-spec/issues/387

File: 05-services.md [hunk 7]
```diff
@@ -901,7 +884,7 @@ healthcheck:
 
 `interval`, `timeout`, `start_period`, and `start_interval` are [specified as durations](11-extension.md#specifying-durations).
```
Comment (aevesdocker) [inline 05-services.md:885]: TODO: https://github.com/docker/docs/issues/13361

File: 05-services.md [hunk 8]
```diff
@@ -1025,8 +1008,8 @@ Compose creates containers with canonical labels:
 - `com.docker.compose.project` set on all resources created by Compose to the user project name
 - `com.docker.compose.service` set on service containers with service name as defined in the Compose file
 
-The `com.docker.compose` label prefix is reserved. Specifying labels with this prefix in the Compose file must
-result in a runtime error.
+The `com.docker.compose` label prefix is reserved. Specifying labels with this prefix in the Compose file
+results in a runtime error.
 
 ### links
```
Comment (aevesdocker) [inline 05-services.md:1014]: https://github.com/compose-spec/compose-spec/issues/394

File: 05-services.md [hunk 9]
```diff
@@ -1025,11 +1012,15 @@ Compose creates containers with canonical labels:
...
 
+> **Note**
```
Comment (aevesdocker) [inline 05-services.md:1020]: @glours would this be enough to clarify? Or shall I still specifically call out the merging of v2andv3 file format?

File: 05-services.md [hunk 10]
```diff
@@ -1209,7 +1197,7 @@ networks:
 `priority` indicates in which order Compose connects the service’s containers to its
 networks. If unspecified, the default value is 0.
 
-In the following example, the app service connects to app_net_1 first as it has the highest priority. It then connects to app_net_3, then app_net_2, which uses the default priority value of 0.
+In the following example, the app service connects to `app_net_1` first as it has the highest priority. It then connects to app_net_3, then `app_net_2`, which uses the default priority value of 0.
```
Comment (glours) [inline 05-services.md:1200]:
```suggestion
In the following example, the app service connects to `app_net_1` first as it has the highest priority. It then connects to `app_net_3`, then `app_net_2`, which uses the default priority value of 0.
```

File: 05-services.md [hunk 11]
```diff
@@ -1306,30 +1295,33 @@ platform: linux/arm64/v8
...
   Compose offers support for platform-specific protocol names.
 
-Host IP, if not set, must bind to all network interfaces. Port can be either a single
+If host IP is not set, it binds to all network interfaces. Ports can be either a single
 value or a range. Host and container must use equivalent ranges.
 
-Either specify both ports (`HOST:CONTAINER`), or just the container port. In the latter case, the
-Compose automatically allocates any unassigned host port.
+Either specify both ports (`HOST:CONTAINER`), or just the container port. In the latter case,
+Compose automatically allocates any unassigned host a port.
```
Comment (glours) [inline 05-services.md:1319]: this the port which is unassigned not the host
```suggestion
Compose automatically allocates any unassigned port of the host.
```

File: 05-services.md (hunk 9 above)
Comment (glours) [reply 05-services.md:1020]: IHMO is good enough, we don't need to details why it was deprecated in Swarm, just let users know some implementations may not support it is enough

**Review by glours — APPROVED** [review]
Few minor changes to fix before merging
Amazing job @aevesdocker, thank you so much 🙏

**Review by ndeloof — APPROVED** [review]

**Pull request merged**